tests/base/src/test/java/org/wildfly/security/ssl/TLS13AuthenticationTest.java

```diff
@@ -114,12 +116,69 @@ public void testTwoWayTLS13() throws Exception {
                 .setNeedClientAuth(true)
                 .build().create();
 
-        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-two-way-tls13.org", "wildfly-ssl-test-config-v1_5.xml", CIPHER_SUITE);
+        SecurityIdentity identity = performConnectionTest(serverContext, "protocol://test-two-way-tls13.org", "wildfly-ssl-test-config-v1_5.xml", CIPHER_SUITE, true);
         assertNotNull(identity);
         assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
     }
 
-    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, String clientConfigFileName, String expectedCipherSuite) throws Exception {
+    @Test
+    public void testDifferentPreferredTLS13Suites() throws Exception {
+        final String REQUIRED_CIPHER_SUITE = "TLS_AES_128_GCM_SHA256";
+        final String PREFERRED_CIPHER_SUITE = "TLS_AES_256_GCM_SHA384";
```

Comment:
Just to check, what's being tested here? Is it the case where a server and client have different sets of TLSv1.3 protocols enabled?

Reply:
Yeah, server and client both prefer a different TLSv1.3 cipher, but also support the required cipher `TLS_AES_128_GCM_SHA256` \([source](https://datatracker.ietf.org/doc/html/rfc8446#section-9.1)\)